#SECOND POST

This is the second post in the series

In [18]:
#this is a thanksgiving data analaysis program, a step to data science
#in this program we want to answer the following question
#what cooking style can be recommended for a particular choice of food for thanksgiving
#what kind of food are people most likely to eat during thanksgiving and how many eat each kind
#what proportion of people celebrate thanksgiving

import pandas as pd
import numpy as np

thanksgiving = pd.read_csv("thanks.csv",engine="python") #use engine=python if encoding error
i_celebrate=thanksgiving[thanksgiving["Do you celebrate Thanksgiving?"]=="Yes"]
i_dont_celebrate=thanksgiving[thanksgiving["Do you celebrate Thanksgiving?"]=="No"]
total_who_celebrate=int(len(i_celebrate) + len(i_dont_celebrate))
percentage_who_dont_celebrate=int((len(i_dont_celebrate)/(total_who_celebrate)*100))
print("Proportion of those that dont celebrate is : ")
print(str(i_dont_celebrate.shape[0]) + " out of " + str(total_who_celebrate) + "persons do not celebrate thanksgiving")
print("Hence " + str(percentage_who_dont_celebrate) + "% do not celebrate thanksgiving" ) 
print("1 out of every " + str(int(len(i_celebrate)/len(i_dont_celebrate))) + " persons are ungrateful")

print("---Question 3 has been answered----")

print("------------****************************question 2************************************-----------------------")

food_choices=thanksgiving["What is typically the main dish at your Thanksgiving dinner?"].unique()
food_choices=list(food_choices)
food_people={}
food_styles={}
for c in food_choices:
    this_c_df=thanksgiving[thanksgiving["What is typically the main dish at your Thanksgiving dinner?"]==c]
    styles_cook=this_c_df["How is the main dish typically cooked?"].unique()
    for style in styles_cook:
        food_style=this_c_df[this_c_df["How is the main dish typically cooked?"]==style]
        num_who_use_this_style=len(food_style)
        food_styles[c +'-'+ style]=num_who_use_this_style
    num_eat_c=this_c_df.shape[0]
    food_people[c]=num_eat_c

#countlist=[]
#for key in food_people:
#    countlist.append(food_people[key])
#countlist.remove(859)
#print(countlist)

print("here is the styles for cooking ")
for i in food_styles:
    print(i  +" --- "+  str(food_styles[i]))




Proportion of those that dont celebrate is : 
78 out of 1058persons do not celebrate thanksgiving
Hence 7% do not celebrate thanksgiving
1 out of every 12 persons are ungrateful
---Question 3 has been answered----
------------****************************question 2************************************-----------------------
here is the styles for cooking 
Ham/Pork-Baked --- 22
Chicken-Fried --- 2
Other (please specify)-Baked --- 16
Tofurkey-I don't know --- 1
Turkey-Other (please specify) --- 34
Turkey-Fried --- 41
Other (please specify)-Roasted --- 5
Roast beef-Baked --- 3
Tofurkey-Roasted --- 4
I don't know-Roasted --- 1
I don't know-I don't know --- 3
Roast beef-Other (please specify) --- 1
Chicken-Baked --- 4
Chicken-Other (please specify) --- 2
Other (please specify)-Other (please specify) --- 12
Chicken-Roasted --- 4
Turkey-Roasted --- 351
Ham/Pork-I don't know --- 1
Turkey-Baked --- 422
Turkey-I don't know --- 11
Tofurkey-Other (please specify) --- 1
I don't know-Fried --- 1
Ham/P

In [7]:
import pandas as pd
data=pd.read_csv("thanksgiving.csv",encoding="Latin-1")
cat_yes_no=data["Do you celebrate Thanksgiving?"].value_counts()
main_dish=data["What is typically the main dish at your Thanksgiving dinner?"]
main_dish_count=data["What is typically the main dish at your Thanksgiving dinner?"].value_counts()
print(cat_yes_no)
print(main_dish_count)
main_dish_tofurkey=data[main_dish=="Tofurkey"]
main_dish_tofurkey_yes_no=main_dish_tofurkey["Do you typically have gravy?"].value_counts()
print(main_dish_tofurkey_yes_no)


Yes    980
No      78
dtype: int64
Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
dtype: int64
Yes    12
No      8
dtype: int64


In [8]:
import pandas as pd
apple=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"]
apple_isnull=pd.isnull(apple)
pumpkin=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"]
pumpkin_isnull=pd.isnull(pumpkin)
pecan=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"]
pecan_isnull=pd.isnull(pecan)
ate_pies=apple_isnull & pumpkin_isnull & pecan_isnull
data["ate_pies"]=ate_pies
cat_ate_pies=data["ate_pies"].unique()
cat_count={}
for c in cat_ate_pies:
    this_c=data[data["ate_pies"]==c]
    cat_count[c]=this_c.shape[0]

print(cat_count)


{False: 876, True: 182}


In [12]:
def return_int_agestring(age_string):
    if(pd.isnull(age_string)):
        return None
    age_string=age_string.split(" ")[0]
    age_string=age_string.replace("+", "")
    
    return int(age_string)
import pandas as pd
data=pd.read_csv("thanksgiving.csv",encoding="Latin-1")
age_integer=data["Age"].apply(return_int_agestring)
data["int_age"]=age_integer
result_described=data["int_age"].describe()
print(result_described)
#print(data["Age"].value_counts())
#print(data["int_age"].value_counts())

count    1025.000000
mean       39.383415
std        15.398493
min        18.000000
25%        30.000000
50%        45.000000
75%        60.000000
max        60.000000
Name: int_age, dtype: float64


In [19]:
def return_intincome(string_income):
    if(pd.isnull(string_income)):
        return None
    
    string_income=string_income.split(" ")[0]
    if(string_income=='Prefer'):
        return None
    string_income=string_income.replace("$", "")
    string_income=string_income.replace(",", "")
    return int(string_income)

earn_count=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts()
print(earn_count)
income_integer=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(return_intincome)
data["int_income"]=income_integer
result_described=data["int_income"].describe()
print(result_described)



$25,000 to $49,999      180
Prefer not to answer    136
$50,000 to $74,999      135
$75,000 to $99,999      133
$100,000 to $124,999    111
$200,000 and up          80
$10,000 to $24,999       68
$0 to $9,999             66
$125,000 to $149,999     49
$150,000 to $174,999     40
$175,000 to $199,999     27
dtype: int64
count       889.000000
mean      74077.615298
std       59360.742902
min           0.000000
25%       25000.000000
50%       50000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64


Findings

Although we only have a rough approximation of income, and it skews downward because we took the first value in each string (the lower bound), the average income seems to be fairly high, although there is also a large standard deviation.

In [21]:
travel_less_15=data[data["int_income"]<150000]
income=travel_less_15["int_income"].values
how_far=travel_less_15["How far will you travel for Thanksgiving?"].value_counts()
print(how_far)


Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
dtype: int64


# INTERPRETATION
i can hypothesize that more number of people earning less stayed at home for their thanksgiving and only few travelled to their parents home for thanksgiving

In [22]:
travel_greater_15=data[data["int_income"]>150000]
income=travel_greater_15["int_income"].values
how_far_greater=travel_greater_15["How far will you travel for Thanksgiving?"].value_counts()
print(how_far_greater)

Thanksgiving is happening at my home--I won't travel at all                         49
Thanksgiving is local--it will take place in the town I live in                     25
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    16
Thanksgiving is out of town and far away--I have to drive several hours or fly      12
dtype: int64


# INTERPRETATION
here the number of those who would travel for thanksgiving is one third of those staying at home for their thanksgiving

In [23]:
data.pivot_table(
    index="Have you ever tried to meet up with hometown friends on Thanksgiving night?",
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_age"
)

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [24]:
data.pivot_table(index='Have you ever attended a "Friendsgiving?"',
                columns="Have you ever tried to meet up with hometown friends on Thanksgiving night?",
                values="int_income")

Have you ever tried to meet up with hometown friends on Thanksgiving night?            No  \
Have you ever attended a "Friendsgiving?"                                                   
No                                                                           78914.549654   
Yes                                                                          72894.736842   

Have you ever tried to meet up with hometown friends on Thanksgiving night?           Yes  
Have you ever attended a "Friendsgiving?"                                                  
No                                                                           78750.000000  
Yes                                                                          66019.736842

#figuring out most common dessert people eat, 
do people eat one or more of:
    Brownies
    Peach cobbler
    Fudge
    Cookies
    Apple cobbler
    Blondies
    Ice cream
    Cheesecake
    Carrot cake
    

In [17]:
import pandas as pd
data=pd.read_csv("thanks.csv",encoding="Latin-1")
Brownies_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies"])
Peach_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler"])
Fudge_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Fudge"])
Cookies_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cookies"])
Apple_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler"])
Blondies_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies"])
Icecream_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream"])
Cheesecake_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake"])
Carrotcake_isnull=pd.isnull(data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Carrot cake"])

dessert_ate_by_all=Brownies_isnull & Peach_isnull & Fudge_isnull & Cookies_isnull & Apple_isnull & Blondies_isnull & Icecream_isnull & Cheesecake_isnull & Carrotcake_isnull
data["dessert_ate_by_all"]=dessert_ate_by_all
data["dessert_ate_by_all"].value_counts()





False    577
True     481
dtype: int64

#from above we can deduce that 577 people would eat one or more of the above dessert and 481 would not eat any of the above dessert. 
#we can say here that if we would recommend dessert during thanksgiving we can fairly recommend any or all of the above where one can make his choice.

#the question would be: what desserts should we recommend during subsequent thanksgiving?


In [24]:
from pandas import Series
brownies=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies"][Brownies_isnull==False]
peach=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler"][Peach_isnull==False]
fudge=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Fudge"][Fudge_isnull==False]
cookies=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cookies"][Cookies_isnull==False]
apple=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler"][Apple_isnull==False]
blondies=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies"][Blondies_isnull==False]
icecream=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream"][Icecream_isnull==False]
cheesecake=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake"][Cheesecake_isnull==False]
carrot=data["Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Carrot cake"][Carrotcake_isnull==False]

print("Below are number of those who ate each of the dessert:")
print("Brownies " + str(len(brownies)))
print("Peach cobbler " + str(len(peach)))
print("Fudge " + str(len(fudge)))
print("Cookies " + str(len(cookies)))
print("Apple cobbler " + str(len(apple)))
print("Blondies " + str(len(blondies)))
print("Ice cream " + str(len(icecream)))
print("Cheese cake " + str(len(cheesecake)))
print("Carrot cake " + str(len(carrot)))

desserts=Series([len(brownies),len(peach),len(fudge),len(cookies),len(apple),len(blondies),len(icecream),len(cheesecake),len(carrot)])
maximum_score=desserts.max()
print("Highest num is " + str(maximum_score))


Below are number of those who ate each of the dessert:
Brownies 128
Peach cobbler 103
Fudge 43
Cookies 204
Apple cobbler 110
Blondies 16
Ice cream 266
Cheese cake 191
Carrot cake 72
Highest num is 266


#most common dessert from the above analysis is
#Ice cream
#hence we can recommend: Ice cream, cookies, cheese cake as top three dessert choice of people during thanksgiving.
#although some people that went for one type of dessert also are likely to go for other type of dessert


#the most common dish people eat can be gotten as:


In [25]:
data["What is typically the main dish at your Thanksgiving dinner?"].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
dtype: int64

Hence Turkey is the most common dish at thanksgiving

#How many people work on thanksgiving
to figure this out, it is dependent on the question
#Will you employer make you work on Black Friday?
#Do you work in retail?


In [30]:
data["Will you employer make you work on Black Friday?"].value_counts()


Yes              43
No               20
Doesn't apply     7
dtype: int64

In [31]:
data["Do you work in retail?"].value_counts()

No     881
Yes     70
dtype: int64

In [34]:
#figuring out people who would work on black friday and those that are retailers or not from then
data_new=data[data["Will you employer make you work on Black Friday?"]=="Yes"]
data_new["Do you work in retail?"].value_counts()

Yes    43
dtype: int64

#from the above, we can analyse that:
#all employers that work on black friday are retailers
#27 people are self employed working as a retailer: since 70 work in retail, 43 will work for an employer on black friday, and all 43 do work in retail. Hence the remaining (70-43) 27 who work in retail do not work for an employer.

#finding regional patterns


In [35]:
data["US Region"].value_counts()

South Atlantic        214
Middle Atlantic       159
East North Central    150
Pacific               146
West South Central     91
West North Central     74
East South Central     60
New England            58
Mountain               47
dtype: int64

In [38]:
#associating regional patterns in dinner menu
#we try to find what kinds of food are eaten by a kind of regional pattern
#this will help us know how to market the dish 

def selectdish(region):
    dish_list=data[data["US Region"]==region]["What is typically the main dish at your Thanksgiving dinner?"]
    return Series(dish_list)

dishlist=selectdish("South Atlantic")
print(dishlist.value_counts())


Turkey                    181
Ham/Pork                    7
Other (please specify)      6
Roast beef                  3
Tofurkey                    3
Chicken                     3
dtype: int64


In [ ]:
#from the above analysis we can say that Turkey is most eaten by south atlantic people
#this will enhance our marketing and awareness of our brand of turkey during thanksgiving

In [59]:
#import numpy as np
dishlist_middleatlantic=selectdish("Middle Atlantic")
print(dishlist_middleatlantic.value_counts())
#data["int_income"].value_counts()
#data.pivot_table(index="What is typically the main dish at your Thanksgiving dinner?",values="int_income",aggfunc=np.sum)

Turkey                    130
Tofurkey                    5
Other (please specify)      4
Roast beef                  2
Ham/Pork                    2
Turducken                   1
Chicken                     1
dtype: int64


In [ ]:
#from the above analysis we can say that Turkey is most eaten by Middle Atlantic people
#this will enhance our marketing and awareness of our brand of turkey during thanksgiving.
#giving from the most consumed dish type above being turkey, it will be obvious that the 9 regions will prefer
#turkey to other type of dish

#now we want to ask this question:
#why do people go more for turkey than the other kinds of dish
#could it be because of income?
#lets relate average income for each type of dish then...

In [57]:
def return_intincome(string_income):
    if(pd.isnull(string_income)):
        return None
    
    string_income=string_income.split(" ")[0]
    if(string_income=='Prefer'):
        return None
    string_income=string_income.replace("$", "")
    string_income=string_income.replace(",", "")
    return int(string_income)

earn_count=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts()
#print(earn_count)
income_integer=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(return_intincome)
data["int_income"]=income_integer
result_described=data["int_income"].describe()
#print(result_described)
#data["int_income"].value_counts()
data.pivot_table(index="What is typically the main dish at your Thanksgiving dinner?",values="int_income",aggfunc=np.mean)

What is typically the main dish at your Thanksgiving dinner?
Chicken                    40500.000000
Ham/Pork                   65370.370370
I don't know               16666.666667
Other (please specify)     79193.548387
Roast beef                 35625.000000
Tofurkey                   73235.294118
Turducken                 200000.000000
Turkey                     77113.543092
Name: int_income, dtype: float64

In [ ]:
#from the above we can see that average income for those who eat Turducken is higher than those who eat Turkey
#hence we musnt conclude that it is because of income, rather it might be choice or likeness for the food
#though it is fine to hypothesize since Turkey is the next after Turducken...hence income could influence this as well

#looking at it based on cost: Turducken might be more expensive than Turkey and 
#Turkey might be expensive than the other kinds of food, hence people went for dish their income could afford

#acording to a source: http://www.wholefoodsmarket.com/blog/turducken-showstopping-holiday-roast
#turducken is a combination of deboned turkey, deboned chicken and deboned duck stuffed into each other and spiced and then
#cooked for about 5-7 hourse...who wouldnt say this might cost a lot, but it taste nice and your thanksgiving would still be talked
#about even after days to come...
